content/news/posts/2024/meet-prestashop9-api.md

```diff
@@ -0,0 +1,386 @@
+
+```json
+{
+    "type": "https://tools.ietf.org/html/rfc2616#section-10",
+    "title": "An error occurred",
+    "status": 403,
+    "detail": "Invalid scope.",
+    "class": "Symfony\\Component\\HttpKernel\\Exception\\AccessDeniedHttpException",
+    "trace": []
+}
+```
+
+## Retrieve a single resource
+
+Now that you have a better understanding of how to authenticate your client and retrieve information about it, you can try to retrieve other resources from the API. You can use the access token you generated earlier to authenticate your client.
+
+In the following example, we will try retrieving and saving user group information. To do that, you need to send a `GET` request to the `/admin-api/customers/group/{customerGroupId}` endpoint, where `{customerGroupId}` is the ID of the user group you want to retrieve. You need to provide the access token in the `Authorization` header.
+
+```bash
+curl --location 'http://yourdomain.test/admin-api/customers/group/1' \
+--header 'Authorization: Bearer YOUR_ACCESS_TOKEN'
+```
+
+If everything goes well, you should receive a response with information about the user group. The response should look like this:
+
+```json
+{
+    "customerGroupId": 1,
+    "localizedNames": {
+        "1": "Odwiedzający",
+        "2": "Visitor"
+    },
+    "reductionPercent": 0.0,
+    "displayPriceTaxExcluded": false,
+    "showPrice": true,
+    "shopIds": [
+        1
+    ]
+}
+```
+
+Notice that you can access the localized names of the user group and all other relevant information.
+
+If you access the endpoint with the wrong ID, you will receive an error message. For example, if you try to access the `/admin-api/customers/group/999999` endpoint, you will receive the following response:
+
+```json
+{
+    "type": "https://tools.ietf.org/html/rfc2616#section-10",
+    "title": "An error occurred",
+    "status": 500,
+    "detail": "Group #4 was not found",
+    "class": "PrestaShop\\PrestaShop\\Core\\Domain\\Customer\\Group\\Exception\\GroupNotFoundException",
+    "trace": []
+}
+```
+
+The endpoint will return a 404 status code and an error message saying the group was not found.
+
+{{% notice success %}}
+If you search for a list of all available resources, you can use the http://yourdomain.test/admin-api/docs.html URL to access the API documentation. The documentation is generated automatically and provides information about all available endpoints, request methods, and parameters.
+{{% /notice %}}
+
+## Editing a single resource
+
+Now that you know how to retrieve information about a single resource, you can try to edit it. To do that, you need to send a `PUT` request to the `/admin-api/customers/group/{customerGroupId}` endpoint, where `{customerGroupId}` is the ID of the user group you want to edit. You need to provide the access token in the `Authorization` header and the data you want to change in the request body.
```

Comment:
Ah it should be a partial update ith PATCH here, but the endpoint wasn't configured in that way We'll have to adapt the module to improve this

Reply:
Should we add a paragraph to explain the difference between PUT and PATCH and that some endpoints will be different depending on their ability to handle partial update?

Reply:
@jolelievre good idea to explain the difference